p2p/node_info.go

```diff
@@ -245,6 +245,18 @@ func (info DefaultNodeInfo) ToProto() *tmp2p.DefaultNodeInfo {
 	return dni
 }
 
+// AddChannel checks if the nodeInfo doesn't already contain the channel before
+// appending it
+func (info DefaultNodeInfo) AddChannel(chID byte) error {
```

Comment:
this not being a pointer receiver makes it seem like the mutation on 256 wouldn't persist outside of the scope of this method.

Having said that it's also probably the case that this general assumption of "pointer methods mutate, and concrete method don't" doesn't hold up well when structs have fields that are references (like slices) which are mutable, even if the outer struct isn't? Which is to say, this might work, but also it's a bit weird.

For clarity I'd make it a pointer.